pkg/lib/packagekit.js

```diff
@@ -404,33 +400,80 @@ export function check_missing_packages(names, progress_cb) {
     }
 
-    return refresh().then(resolve)
+    return refresh(false, progress_cb).then(resolve)
             .then(simulate)
             .then(get_details);
 }
 
-/* Check a list of packages whether they are installed.
+/**
+ * Check a list of packages whether they are installed.
  *
- * This is a lightweight version of check_missing_packages() which does not
- * refresh, simulates, or retrieves details. It just checks which of the given package
- * names are already installed, and returns a Set of the missing ones.
+ * @param {string[]} names - names of packages which should be installed
+ * @return {Promise<boolean>} true if packages are installed
  */
-export function check_uninstalled_packages(names) {
+export async function is_installed(names, progress_cb) {
     const uninstalled = new Set(names);
 
-    if (names.length === 0)
-        return Promise.resolve(uninstalled);
+    if (uninstalled.size === 0)
+        return true;
 
-    return cancellableTransaction("Resolve",
-                                  [Enum.FILTER_ARCH | Enum.FILTER_NOT_SOURCE | Enum.FILTER_INSTALLED, names],
-                                  null, // don't need progress, this is fast
-                                  {
-                                      Package: (info, package_id) => {
-                                          const parts = package_id.split(";");
-                                          uninstalled.delete(parts[0]);
-                                      },
-                                  })
-            .then(() => uninstalled);
+    await cancellableTransaction("Resolve",
+                                 [Enum.FILTER_ARCH | Enum.FILTER_NOT_SOURCE | Enum.FILTER_INSTALLED, names],
+                                 progress_cb,
+                                 {
+                                     Package: (info, package_id) => {
+                                         const parts = package_id.split(";");
+                                         uninstalled.delete(parts[0]);
+                                     },
+                                 });
+
+    return uninstalled.size === 0;
+}
+
+/**
+ * Check a list of packages whether they are available.
+ *
+ * @param {string[]} names - names of packages which should be available in the repositories
+ * @return {Promise<boolean>} true if packages are available
+ */
+export async function is_available(names, progress_cb) {
+    const available = new Set();
+
+    if ((names && names.length === 0))
+        return true;
+
+    await cancellableTransaction("Resolve",
+                                 [Enum.FILTER_ARCH | Enum.FILTER_NEWEST | Enum.FILTER_NOT_INSTALLED, names],
+                                 progress_cb,
+                                 {
+                                     Package: (info, package_id) => {
+                                         const pkgname = package_id.split(";")[0];
+                                         available.add(pkgname);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22498-d7289306-20251119-185431-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)